File: mobile/app/src/main/java/com/geneirodan/nextgenmeal/data/requests/OrderRequest.kt
package com.geneirodan.nextgenmeal.data.requests

import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable

@Serializable
data class OrderRequest(
    val time: LocalDateTime,
    val isBox: Boolean,
    val cateringId: Int,
    val orderDishes: List<OrderDishRequest>
)

File: server/API/Requests/OrderRequest.cs
using System.ComponentModel.DataAnnotations;

namespace API.Requests;

public class OrderRequest : IRequestBody
{
    [Required]
    [DataType(DataType.DateTime)]
    public DateTime Time { get; set; }
    [Required]
    public bool? IsBox { get; set; } = null!;
    [Required]
    public int CateringId { get; set; }
    [Required]
    public List<OrderDishRequest> OrderDishes { get; set; } = null!;
}
